shared/actions/config/index.js

```diff
@@ -67,7 +56,25 @@ export function getExtendedConfig (): AsyncAction {
   }
 }
 
-export function getCurrentStatus (): AsyncAction {
+let bootstrapSetup = false
+export function bootstrap (): AsyncAction {
+  return dispatch => {
+    if (!bootstrapSetup) {
+      bootstrapSetup = true
+      console.log('Registered bootstrap')
+      engine.listenOnConnect('bootstrap', () => {
+        console.log('Bootstrapping')
+        dispatch(bootstrap())
+      })
+    } else {
+      Promise.all([dispatch(getCurrentStatus()), dispatch(getExtendedStatus()), dispatch(getConfig())]).then(() => {
```

Comment:
It would be nice if this were in a separate function that just returns `.all` call and catches and logs errors. Then here would just call that and `.then(() => dispatch(navBasedOnLoginState()))`